### PR TITLE
fix(dataframe): correct schema of search result in DataFrame.apply

### PR DESCRIPTION
**Pull request by AChenQ**

### CODE REVIEW

File: graviti/dataframe/frame.py [hunk 1]
```diff
@@ -832,9 +832,11 @@ def apply(self, func: Callable[[Any], Any]) -> Container:
         if self.searcher is None:
             raise TypeError("'apply' is not supported for the DataFrame not in a Commit")
 
+        key = "result"
         result = func(SqlRowSeries(self.schema))
         criteria = {} if self.criteria is None else self.criteria.copy()
-        criteria["select"] = [result.expr]
-        dataframe = self.searcher(criteria)  # pylint: disable=not-callable
-        key = dataframe._column_names[0]  # pylint: disable=protected-access
+        criteria["select"] = [{key: result.expr}]
```
Comment (linjiX) [inline graviti/dataframe/frame.py:838]: shall we move this into another PR?

**Review by linjiX — APPROVED** [review]

**Review by graczhual — APPROVED** [review]
